### PR TITLE
Fixed proximity calculations in proximityeffect and added tests

### PR DESCRIPTION
**Pull request by thalbern**
## Overview
- fixed problem when proximity ranges are bigger than interaction radius of near pointers
- fixed wrong calculation of proximity range around the box to determine which pointers are in range
- changed comparisons for proximity to less-equal
- wrote tests that confirm 
-- default behavior proximity effect switched off
-- default configuration on proximity effect is working when switched on
-- user configured confirmation during runtime is applied and working

- minor: moved all bounds control scriptable configs into experimental menu entries

## Changes
- Fixes: #5340 - task: add tests for proximity effect

### CODE REVIEW

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/ProximityEffect/ProximityEffect.cs [hunk 1]
```diff
@@ -137,30 +137,34 @@ public void UpdateScaling(Vector3 boundsCenter, Vector3 boundsExtents)
...
                     }
                 }
             }
 
-            // Get the max radius possible of our current bounds plus the proximity
-            float maxRadius = Mathf.Max(Mathf.Max(boundsExtents.x, boundsExtents.y), boundsExtents.z);
-            maxRadius *= maxRadius;
-            maxRadius += config.ObjectCloseProximity + config.ObjectMediumProximity;
+            // Get the max radius possible of our current bounds and extent the range to include proximity scaled objects
+            float squareMaxLength = boundsExtents.sqrMagnitude + (3 * config.ObjectMediumProximity * config.ObjectMediumProximity);
```
Comment (MenelvagorMilsom) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/ProximityEffect/ProximityEffect.cs]: Where does this 3 come from? Is it that `(3 * config.ObjectMediumProximity * config.ObjectMediumProximity)` is equivalent to the square magnitude of a vector where xyz are all `config.ObjectMediumProximity`?

Comment (thalbern) [reply Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/ProximityEffect/ProximityEffect.cs]: yes exactly that! I'm expanding the existing bounds by the medium proximity value. i'll add a comment

File: Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/ProximityEffect/ProximityEffect.cs [hunk 2]
```diff
@@ -137,30 +137,34 @@ public void UpdateScaling(Vector3 boundsCenter, Vector3 boundsExtents)
...
                         proximityPointers.Add(pointer);
                     }
                 }
             }
 
-            // Get the max radius possible of our current bounds plus the proximity
-            float maxRadius = Mathf.Max(Mathf.Max(boundsExtents.x, boundsExtents.y), boundsExtents.z);
-            maxRadius *= maxRadius;
-            maxRadius += config.ObjectCloseProximity + config.ObjectMediumProximity;
+            // Get the max radius possible of our current bounds and extent the range to include proximity scaled objects
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/ProximityEffect/ProximityEffect.cs]:
```suggestion
            // Get the max radius possible of our current bounds and extent the range to include proximity scaled objects. This is done by adjusting the original bounds to include the ObjectMediumProximity range in x, y and z axis
```

**Pull request merged**